x-pack/plugins/endpoint/server/routes/alerts.test.ts

```diff
@@ -0,0 +1,191 @@
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.total).toEqual(132);
+    expect(alertResultList.request_page_index).toEqual(0);
+    expect(alertResultList.request_page_size).toEqual(10);
+  });
+
+  it('should return alert results according to pagination params -- POST', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      method: 'post',
+      body: {
+        paging_properties: [
+          {
+            page_size: 20,
+          },
+          {
+            page_index: 2,
+          },
+        ],
+      },
+    });
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() =>
+      Promise.resolve((data as unknown) as SearchResponse<AlertData>)
+    );
+    [routeConfig, routeHandler] = routerMock.post.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/alerts')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalled();
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.alerts.length).toEqual(20);
+    expect(alertResultList.total).toEqual(132);
+    expect(alertResultList.request_page_index).toEqual(40);
+    expect(alertResultList.request_page_size).toEqual(20);
+  });
+
+  it('should return alert results according to pagination params -- GET', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      path: '/api/endpoint/alerts?page_size=20&page_index=2',
+      query: {
+        page_size: 20,
+        page_index: 2,
+      },
+    });
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() =>
+      Promise.resolve((data as unknown) as SearchResponse<AlertData>)
+    );
+    [routeConfig, routeHandler] = routerMock.get.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/alerts')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalled();
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.alerts.length).toEqual(20);
+    expect(alertResultList.total).toEqual(132);
+    expect(alertResultList.request_page_index).toEqual(40);
```

Comment:
why would `request_page_index` be 40 here?

Reply:
It's really the cursor index, not the page index. These params work analogous to the Endpoint Management APIs and there are a few things I'd like to change about them. Namely that the page_index starts at 0 instead of 1, and that the semantics of the `page_index` are different on input vs output.

Reply:
FYI - @parkiino and I ran across this recently as well and had a discussion with @nnamdifrankie on it. He has an open issue to come back and refactor it. see: elastic/endpoint-app-team/issues/135

Reply:
Added `result_from_index` and adjusted value of `request_page_index`. Resolving.